edb/lib/ext/auth.edgeql

```diff
@@ -76,6 +76,16 @@ CREATE EXTENSION PACKAGE auth VERSION '1.0' {
             create constraint exclusive;
         };
 
+        create trigger email_shares_user_handle after insert for each do (
+            std::assert(
+                (__new__.user_handle = (
+                    select detached ext::auth::WebAuthnFactor
+                    filter .email = __new__.email
+                    and not .id = __new__.id
+                ).user_handle) ?? true,
```

Comment:
Minor thing, but I think `?? true` probably isn't needed. If the result of the select is an empty set, the assertion just won't run anyway.

Reply:
TIL about the empty set behavior of `assert`. I'll add an example to the documentation for `assert`.

Reply:
I think it just follows the normal set behaviour, where if the function arg isn't `set of` (like in `count`, for example) then it's expanded out. So `assert({true, false})` is executed as `{assert(true), assert(false)}`, and similarly `assert({})` becomes `{}`.

Reply:
That makes sense. The documentation hints at this to since it says:

> If the input bool is false, assert raises a QueryAssertionError. _Otherwise, this function returns true._ (emphasis mine)

I think it'd just be helpful to show some examples of calling this function with a sets to show that. I'll follow up.